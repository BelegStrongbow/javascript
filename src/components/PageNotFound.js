import React from "react";
import pageNotFoundImage from "../images/PageNotFound.svg";
import { defineMessages, injectIntl, intlShape, FormattedMessage } from "react-intl";
import a11ySpeak from "a11y-speak";
import styled from "styled-components";
import { Heading } from "./Headings";

const PageContainer = styled.div`
	margin-top: 10vh;
	text-align: center;

	img {
		width: 384px;
		max-width: 100%;
		height: auto;
		margin-top: 2em;
	}
`;

const messages = defineMessages( {
	pageNotFound: {
		id: "page.notfound",
		defaultMessage: "Page not found",
	},
} );

/**
 * A function that returns the Page Not Found component.
 *
 * @returns {ReactElement} The component that contains the Page Not Found page.
 */
class PageNotFound extends React.Component {
	/**
	 * Sets the CoursesPage object.
	 *
	 * @returns {void}
	 */
	constructor() {
		super();
	}

	componentDidMount() {
		// Announce navigation to assistive technologies.
		let message = this.props.intl.formatMessage( messages.pageNotFound );
		a11ySpeak( message );
	}

	render() {
		let paragraphs = [
			<FormattedMessage id="page.notfound" defaultMessage="I'm afraid the page you are looking for does not exist..." />,
		];
		return (


		<PageContainer>
			<Heading><FormattedMessage id="page.notfound.header" defaultMessage={ "Oops" } /></Heading>
			{ paragraphs.map( function( paragraph ) {
				return <p key={ paragraph.props.id }>{ paragraph }</p>;
			} ) }
			<img src={ pageNotFoundImage } alt="" />
		</PageContainer>
		);
	}
}

export default injectIntl( PageNotFound );

PageNotFound.propTypes = {
	intl: intlShape.isRequired,
};
