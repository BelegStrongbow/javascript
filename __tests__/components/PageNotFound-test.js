import React from 'react';
import PageNotFound from '../../src/components/PageNotFound';
import { createComponentWithIntl } from "../../utils";

test('the page not found component matches the snapshot', () => {
	const component = createComponentWithIntl(
		<PageNotFound />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
