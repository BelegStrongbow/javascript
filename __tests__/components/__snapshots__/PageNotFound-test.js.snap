// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the page not found component matches the snapshot 1`] = `
.c1 {
  font-weight: 700;
  font-size: 2em;
}

.c0 {
  margin-top: 10vh;
  text-align: center;
}

.c0 img {
  width: 384px;
  max-width: 100%;
  height: auto;
  margin-top: 2em;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    <span>
      Oops
    </span>
  </h1>
  <p>
    <span>
      I'm afraid the page you are looking for does not exist...
    </span>
  </p>
  <img
    alt=""
    src="PageNotFound.svg"
  />
</div>
`;
